libs/ob-sql-parser/src/main/resources/obmysql/sql/OBParser.g4

```diff
@@ -1041,6 +1120,10 @@ data_type
     | json_type_i
     | collection_type_i
     | geo_type_i
+    | vector_type_i
+    | roaringbitmap_type_i
+    | ARRAY LeftParen data_type RightParen
```

Comment:
`ARRAY LeftParen data_type RightParen` and `data_type LeftBracket RightBracket` may reduce to `array_type_i`

Reply:
Antlr4 is not able to handle with `data_type: array_type_i; array_type_i: ARRAY LeftParen data_type RightParen | data_type LeftBracket RightBracket` because antlr4 can not handle with the **indirect left recursion** but it can handle with the `direct left recursion`.

Reply:
😂，I forget this~